third_party/nvfuser/python/__init__.py

```diff
@@ -2,26 +2,88 @@
 from . import _C
 from ._C import *
 
-class FusionDefinition(_C._FusionDefinition):
+import numbers
+from functools import partial
+
+
+class PythonOperators:
+    def _not_number_scalar(self, num):
+        return not isinstance(num, numbers.Number) or isinstance(num, bool)
+
+    def _convert_python_number(self, op, *args, **kwargs):
+        # TODO: handle python containers as well.
+        args = [
+            x if self._not_number_scalar(x) else self._fd.define_constant(x)
+            for x in args
+        ]
+        for k, v in kwargs.items():
+            if not self._not_number_scalar(v):
+                kwargs[k] = self._fd.define_constant(v)
+        return op(*args, **kwargs)
+
+    def bindOperation(self, op_name: str, op):
+        # TODO remove this after we can handle python containers.
+        if op_name not in ["var", "var_mean"]:
+            setattr(self, op_name, partial(self._convert_python_number, op))
+        else:
+            setattr(self, op_name, op)
+
+    def __init__(self, fd):
+        self._fd = fd
+
+
+class FusionDefinition:
+    Operators = _C._FusionDefinition.Operators
+
+    def __init__(self, id=None, *, max_length=None):
+        self._fd = _C._FusionDefinition(id)
+        self.ops = PythonOperators(self._fd)
+
+        # mapping Operators for query
+        self.Operator = self._fd.Operators
+
+        # map _C._FusionDefinition methods
+        map_methods = {
+            "_setup_definition": "setup_definition",
+            "_finalize_definition": "finalize_definition",
```

Comment:
Why rename these? If you don't rename them, can't you still call `self._fd._setup_definition()`?

Reply:
I can keep them as-is. But then I'll have to keep a wrapper function with `setup_definition()` that just calls into `self._fd._setup_definition()`, since our current API does that.

Just being lazy and try to write less code.